### ISSUE TITLE
`System.GC.Server` setting in `runtimeconfig.template.json` is ignored for ASP.NET Core 6.0

### ISSUE THREAD

**Issue by ozziepeeps** [dotnet/sdk#26497]

**Comment by fmg-lydonchandra** [comment]
One of workarounds is to use MSBuild property `<ServerGarbageCollection>false</ServerGarbageCollection>`

**Comment by ozziepeeps** [comment]
Yes that’s the workaround I went with, but would prefer to have the setting in the runtimeconfig as it’s more visible IMO. Thanks!

**Comment by baronfel** [comment]
This is happening because the default ASP.NET Props set `ServerGarbageCollection` to true invisibly on your behalf [here](https://github.com/dotnet/sdk/blob/72dec847de8ec8cb676be5c837ccbb7ac3a88ab0/src/WebSdk/ProjectSystem/Targets/Microsoft.NET.Sdk.Web.ProjectSystem.props#L20). That then leads to the SDK setting the property to true [here](https://github.com/dotnet/sdk/blob/14a527b419149545e4806ec340e50f9a97bbe3ac/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets#L410-L412), which is then used to overwrite your template [here](https://github.com/dotnet/sdk/blob/4ac47e3bad321eeee87f683c3c041e6557a254c2/src/Tasks/Microsoft.NET.Build.Tasks/GenerateRuntimeConfigurationFiles.cs#L288-L291).

I'm not sure there's a way to fix this other than explicitly setting the MSBuild property to `false`. Your template value is actually used (user templates are applied first, then these MSBuild-derived settings), it's just being immediately overwritten by the ASP.NET SDK.

**Comment by rhousham** [comment]
Not really the same issue but recently had an issue with VS2025 and this. What happened was that I mistyped the name for "runtimeconfig.template.json". I updated it but it didn't update my "projectName.runtimeconfig.json" - I deleted the obj folder - rebuilt the solution and then all good.
